x-pack/test/security_solution_endpoint/apps/endpoint/endpoint_transform.ts

```diff
@@ -0,0 +1,147 @@
+          wait_for_active_shards: 1,
+          mappings: {
+            properties: {
+              '@timestamp': { type: 'date' },
+              uuid: { type: 'keyword' },
+              agent: { type: 'keyword' },
+              b_value: { type: 'keyword' },
+            },
+          },
+        });
+        await esClient.transform.putTransform({
+          transform_id: transformName,
+          dest: {
+            index: transformDest,
+          },
+          source: {
+            index: [indexA, indexB],
+          },
+          frequency: '1s',
+          sync: {
+            time: {
+              delay: '3s',
+              field: '@timestamp',
+            },
+          },
+          pivot: {
+            aggs: {
+              join: {
+                scripted_metric: transformAggregation,
+              },
+            },
+            group_by: {
+              uuid: {
+                terms: {
+                  field: 'uuid',
+                },
+              },
+            },
+          },
+        });
+      });
+      after(async () => {
+        await endpointTestResources.stopTransform(transformName);
+        await esClient.transform.deleteTransform({
+          transform_id: transformName,
+          force: true,
+          delete_dest_index: true,
+        });
+        await esClient.indices.delete({ index: indexA });
+        await esClient.indices.delete({ index: indexB });
+      });
+
+      it('runs scripted metric agg to join data', async () => {
+        await esClient.index({
+          index: indexA,
+          refresh: 'wait_for',
+          document: {
+            '@timestamp': new Date().getTime(),
+            uuid: '8000',
+            Endpoint: 'e', // Endpoint key must be present. value does not matter
+            a_value: 'A', // key to test for
+          },
+        });
+        await esClient.index({
+          index: indexB,
+          refresh: 'wait_for',
+          document: {
+            '@timestamp': new Date().getTime(),
+            uuid: '8000',
+            agent: 'a', // agent key must be present
+            b_value: 'B', // key to test for
+          },
+        });
+        await endpointTestResources.startTransform(transformName);
+
+        await sleep(1000);
+
+        await retry.waitForWithTimeout('transform to run', 9000, async () => {
+          const search = await esClient.search({
+            index: transformDest,
+            query: {
+              match_all: {},
+            },
+            rest_total_hits_as_int: true,
+          });
+          expect(search.hits.total).to.be(1);
+          expect(search.hits.hits[0]._source).to.have.key('join');
+          expect(search.hits.hits[0]._source.join).to.have.key('endpoint');
+          expect(search.hits.hits[0]._source.join).to.have.key('agent');
+          expect(search.hits.hits[0]._source.join.endpoint.a_value).to.be('A');
+          expect(search.hits.hits[0]._source.join.agent.b_value).to.be('B');
```

Comment:
Nit: If this test gets any bigger, could we use strings instead? `const [indexAValue, indexBValue] = ['A', 'B']` would be easier to follow since we can click on them in our IDEs instead of using Ctrl+F to search for "A" :)

Reply:
famous last words, but I don't expect this test file to ever be touched again until we delete our usage of transforms (which is on the agenda)